File: java/obfuscation/add_elements_obf2.java
class SumVariants {
    static int addElements(int[] a) {
        int total = 0;
        int idx = 0;
        while (idx < a.length) {
            int val = a[idx];
            if (val == 0) { /* noop */ }
            total = (total + val) - 0;
            idx++;
        }
        boolean debug = false;
        if (debug) System.out.println("Debug mode enabled");
        return total;
    }
}

File: python/obfuscation/add_elements_obf2.py
def add_elements(nums):
    accumulator = 0
    idx = 0
    while idx < len(nums):
        val = nums[idx]
        if val == 0:
            pass
        accumulator = (accumulator + val) - 0
        idx += 1
    debug = False
    if debug:
        print("Debug mode enabled")
    return accumulator
